azurerm/internal/services/synapse/synapse_sql_pool_resource.go

```diff
@@ -113,7 +112,7 @@ func resourceArmSynapseSqlPool() *schema.Resource {
 				Optional:     true,
 				Computed:     true,
 				ForceNew:     true,
-				ValidateFunc: postgresValidate.PostgresDatabaseCollation,
+				ValidateFunc: mssqlValidate.MsSqlDBCollation(),
```

Comment:
Is this collation specific to only synapse sql pool's or is it generic for all msSql databases? If it is for only synapse I would update the name to be reflective of that, something like `synapseSqlPoolDBCollation`